internal/services/consumer.go

```diff
@@ -71,8 +72,15 @@ func (cgh *consumerGroupHandler) Cleanup(sarama.ConsumerGroupSession) error {
 
 func (cgh *consumerGroupHandler) ConsumeClaim(session sarama.ConsumerGroupSession, claim sarama.ConsumerGroupClaim) error {
 	for message := range claim.Messages() {
-		log.Printf("Message received: value=%s, partition=%d, offset=%d", string(message.Value), message.Partition, message.Offset)
+		cm := jsonToCanaryMessage(message.Value)
+		log.Printf("Message received: value=%+v, partition=%d, offset=%d", cm, message.Partition, message.Offset)
 		session.MarkMessage(message, "")
 	}
 	return nil
 }
+
+func jsonToCanaryMessage(value []byte) CanaryMessage {
+	var cm CanaryMessage
+	json.Unmarshal(value, &cm)
```

Comment:
any use case where we care about error handling the unmarshalling?

Reply:
we are actually generating our own message, so it should be a valid JSON at the source/producer.